File: JeMPI_Apps/JeMPI_UI/src/types/Notification.ts
export default interface Notification {
  id: string
  type: NotificationType
  reason?: string | null
  created: Date
  names: string
  patient_id: string | null
  status: NotificationState
  old_golden_id: string
  current_golden_id: string
  score: number
  linkedTo?: GoldenRecordCandidate
  candidates?: GoldenRecordCandidate[]
}

export interface GoldenRecordCandidate {
  golden_id: string
  score: number
}

export enum NotificationState {
  OPEN = 'OPEN',
  CLOSED = 'CLOSED',
  ALL = 'ALL'
}

export type NotificationType = 'THRESHOLD' | 'MARGIN' | 'UPDATE'

export type Notifications = {
  records: Notification[]
  pagination: {
    total: number
  }
}

File: JeMPI_Apps/JeMPI_LibShared/src/main/java/org/jembi/jempi/shared/models/Notification.java
package org.jembi.jempi.shared.models;

import java.util.List;

public record Notification(
      Long timeStamp,
      // UTC Time
      NotificationType notificationType,
      String dID,
      // Document ID
      String patientNames,
      MatchData linkedTo,
      List<MatchData> candidates) {
   public enum NotificationType {
      THRESHOLD("Threshold"),
      MARGIN("Margin"),
      UPDATE("Update");

      public final String label;

      NotificationType(final String label) {
         this.label = label;
      }
   }

   public enum NotificationState {
      OPEN("OPEN"),
      CLOSED("CLOSED");
      public final String label;

      NotificationState(final String label) {
         this.label = label;
      }
   }

   public record MatchData(
         String gID,
         // Golden ID
         Float score) {
   }
}
